src/utils/auction.ts

```diff
@@ -0,0 +1,169 @@
+import {
+  AuctionData,
+  BackstopToken,
+  FixedMath,
+  Pool,
+  PoolEvent,
+  PoolEventType,
+  PoolOracle,
+} from '@blend-capital/blend-sdk';
+export enum AuctionType {
+  Liquidation = 0,
+  BadDebt = 1,
+  Interest = 2,
+}
+
+export interface AuctionDisplay {
+  auctionData: AuctionData;
+  auctionType: AuctionType;
+  user: string;
+  bidValue: number;
+  lotValue: number;
+  timestamp: string;
+  filled: boolean;
+  blockFilled?: number;
+}
+
+export function processAuctionEvents(
+  events: PoolEvent[],
+  poolOracle: PoolOracle,
+  pool: Pool,
+  backstopToken: BackstopToken,
+  currLedger: number
+): AuctionDisplay[] {
+  let auctions: AuctionDisplay[] = [];
+  for (const event of events ?? []) {
+    switch (event.eventType) {
+      case PoolEventType.NewLiquidationAuction:
+        auctions.unshift({
+          auctionData: event.auctionData,
+          user: event.user,
+          auctionType: AuctionType.Liquidation,
+          bidValue: 0,
+          lotValue: 0,
+          timestamp: event.ledgerClosedAt,
+          filled: false,
+        });
+        break;
+      case PoolEventType.NewAuction:
+        auctions.unshift({
+          auctionData: event.auctionData,
+          user: pool.config.backstop ?? '',
+          auctionType: event.auctionType as AuctionType,
+          bidValue: 0,
+          lotValue: 0,
+          timestamp: event.ledgerClosedAt,
+          filled: false,
+        });
+        break;
+      case PoolEventType.FillAuction:
+        const index = auctions.findIndex(
+          (auction) => auction.user === event.user && auction.auctionType === event.auctionType
+        );
+        if (index !== -1) {
+          let auction = auctions[index];
+          if (event.fillAmount < BigInt(100)) {
+            auction.auctionData = scaleAuction(
+              auction.auctionData,
+              event.ledger,
+              Number(event.fillAmount)
+            );
+          } else {
+            auction.auctionData = scaleAuction(auction.auctionData, event.ledger, 0);
+            auction.filled = true;
+            auction.blockFilled = event.ledger;
+          }
+          auctions[index] = auction;
+        }
+        break;
```

Comment:
We should create unique entries for each fill, otherwise I don't think this tracks correctly. (also, do events include the TX they originated from? It would be nice to include the hash here)

Consider the following:

Auction gets filled at block 200 at 50%, then at block 300 for the remaining 50%.

What is occurring here if I follow, is that the only the fill on block 300 will be represented.

We should create a new `ScaledAuction` that is filled, for each fill operation found, and if any is remaining, include the auction again without being marked as filled with the remaining auction